### PR TITLE
docs(puppet manager): rewrite

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes

- Rewrite `puppet` manager docs:
  - One sentence per line
  - Capitalize headings properly
  - Use Material for MkDocs code block titles
  - Capitalize comments in code blocks
  - Improve "how it works" list

## Context

Follow up after:

- #14789

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/puppet/readme.md [hunk 1]
```diff
@@ -1,78 +1,74 @@
...
+```ruby title="Git-based forge"
+# Tag based
+mod 'stdlib',
     :git => 'git@gitlab.com:example/project_stdlib.git',
     :tag => '5.0.0'
-   ```
+```
 
-### possible improvements
+### Possible improvements
```
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/modules/manager/puppet/readme.md:64]: Generally it's best to have them in issues, not in text

[the remark this answers is not in the document]
Comment (HonkingGoose) [reply lib/modules/manager/puppet/readme.md:64]: I don't really understand the point of the `### Possible improvements` and `#### Further Git-support` sections as they are now.

We show a `# Tag based` `Git-based forge` above, and have a `#Branch based` and `#Ref based` Git example below.

To me it almost reads as if we can drop the `### Possible improvements` and `#### Further Git-support` sections, and move the `# Branch based` and `# Ref based` examples into the `### Supported Puppetfile formats` section.

**Review by rarkins — DISMISSED** [review]

File: lib/modules/manager/puppet/readme.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (rarkins) [reply lib/modules/manager/puppet/readme.md:64]: Yes, let's drop

File: lib/modules/manager/puppet/readme.md [hunk 2]
```diff
@@ -1,85 +1,71 @@
...
 1. Renovate resolves the dependency on the provided forges (or uses `https://forgeapi.puppetlabs.com` as default)
 1. A PR is created with `Puppetfile` updated in the same commit
 1. If the source repository has either a "changelog" file or uses GitHub releases, then Release Notes for each version will be embedded in the generated PR
 
-### supported Puppetfile formats
+### Supported Puppetfile formats
 
-the manager extracts the deps from one Puppetfile
+The `puppetfile` manager extracts the dependencies from one Puppetfile.
+You can define a forge in your `puppetfile` in these ways:
```
Comment (MindTooth) [inline lib/modules/manager/puppet/readme.md:14]: Is the managers name `puppetfile` or `puppet`?

https://docs.renovatebot.com/modules/manager/puppet/

File: lib/modules/manager/puppet/readme.md [hunk 3]
```diff
@@ -1,78 +1,74 @@
...
+mod 'example/infra', '3.3.0'
+```
 
-5. git based version
+```ruby title="GitHub-based forge"
+# Tag based
+mod 'example/standalone_jar',
+    :git => 'git@gitlab.example.de:puppet/example-standalone_jar',
+    :tag => '0.9.0'
+```
```
[the remark this answers is not in the document]
Comment (MindTooth) [reply lib/modules/manager/puppet/readme.md]: Seems Ruby uses two spaces for indent.  I use 2 spaces myself all the time.

https://rubystyle.guide/#spaces-indentation

https://github.com/rubocop/rspec-style-guide#empty-lines-inside-example-group (no spec, but example uses two)

**Review by rarkins — APPROVED** [review]

**Pull request merged**